### PR TITLE
Composer plugin

### PR DESCRIPTION
**Pull request by thomasvargiu**
Developed as composer-plugin.

- auto-configure project on installation with:
  - writing out `.php_cs.dist`
  - adding script commands in project's `composer.json`

Need more tests

### CODE REVIEW

File: src/Installer/Plugin.php [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+     * Return this package name.
+     *
+     * @return string
+     *
+     * @throws \RuntimeException
+     * @throws \InvalidArgumentException
+     */
+    public static function getPackageName(): string
+    {
+        $composerJson = new JsonFile(__DIR__.'/../../composer.json');
```
Comment (Jean85) [inline src/Installer/Plugin.php]: Should be replaced with `dirname(__DIR__, 2) . '/composer.json'`

File: src/Installer/Provider/SourcePaths/ComposerAutoloadProvider.php [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    {
+        $paths = [];
+        $allowedAutoloads = ['psr-0', 'psr-4'];
+        foreach ($this->autoload as $autoloadType => $autoload) {
+            if (! in_array($autoloadType, $allowedAutoloads, true)) {
+                continue;
+            }
+
+            if (! is_array($autoload)) {
+                throw new \RuntimeException('Your composer autoloader section does not contain an array!');
```
Comment (Jean85) [inline src/Installer/Provider/SourcePaths/ComposerAutoloadProvider.php]: Shouldn't this be an `\InvalidArgumentException`?

File: src/DefaultRules.php [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+        ],
+        'not_operator_with_successor_space' => true,
+        'binary_operator_spaces' => [
+            'align_double_arrow' => false,
+            'align_equals' => false,
+        ],
+        'blank_line_after_namespace' => true,
+        'blank_line_after_opening_tag' => true,
+        'cast_spaces' => true,
+        'concat_space' => true,
```
Comment (Jean85) [inline src/DefaultRules.php]: The `concat_space` rule defaults to `none`, I would prefer one space. Is this intentional?

File: src/DefaultRules.php [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+            'syntax' => 'short',
+        ],
+        'not_operator_with_successor_space' => true,
+        'binary_operator_spaces' => [
+            'align_double_arrow' => false,
+            'align_equals' => false,
+        ],
+        'blank_line_after_namespace' => true,
+        'blank_line_after_opening_tag' => true,
+        'cast_spaces' => true,
```
Comment (Jean85) [inline src/DefaultRules.php]: Idem here: it defaults to one space, I would prefer one.

File: src/Installer/Provider/SourcePaths/ComposerAutoloadProvider.php (hunk 2 above)
Comment (thomasvargiu) [reply src/Installer/Provider/SourcePaths/ComposerAutoloadProvider.php]: It's not an argument of the function. If it's not an array means that something's wrong with the `composer.json`

**Pull request merged**